src/screen/ConnectionsList/DeleteConnectionModal/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { FormHandles } from '@unform/core'
+import { Form } from '@unform/web'
+import React, { useRef, useCallback, memo } from 'react'
+import { FiTrash } from 'react-icons/fi'
+import { useToggle } from 'react-use'
+import { useSetRecoilState } from 'recoil'
+
+import { IConnection, connectionsState } from '../../../atoms/connections'
+import Button from '../../../components/Button'
+import Input from '../../../components/Form/Input'
+import Modal, { SharedModalProps } from '../../../components/Modal'
+import { useToast } from '../../../context/toast'
+import { deleteAndGetConnections } from '../../../services/connection/DeleteConnectionService'
+import { TextContent, ActionsContainer, ButtonGroup } from './styles'
+
+interface ConnectionModalProps extends SharedModalProps {
+  connectionToDelete: IConnection
+}
+
+const DeleteConnectionModal: React.FC<ConnectionModalProps> = ({
+  visible,
+  onRequestClose,
+  connectionToDelete
+}) => {
+  const formRef = useRef<FormHandles>(null)
+  const { addToast } = useToast()
+  const setConnections = useSetRecoilState(connectionsState)
+
+  const [deleteConnectionLoading, toggleDeleteConnectionLoading] = useToggle(
+    false
+  )
+
+  const handleCloseModal = useCallback(() => {
+    if (onRequestClose) {
+      onRequestClose()
+    }
+  }, [onRequestClose])
+
+  const handleDeleteConnection = useCallback(async () => {
+    try {
+      toggleDeleteConnectionLoading()
+
+      formRef.current?.setErrors({})
+
+      const { confirm } = formRef.current?.getData() as { confirm: string }
```

Comment:
You can get the form data from the function arguments instead of calling `getDate` from Unform as it's the function inside form `onSubmit`:
```js
const handleDeleteConnection = useCallback(async (data: DeleteConnectionFormData) => {
  console.log(data);
}, []);
```